lib/veri_traci.ml

```diff
@@ -0,0 +1,89 @@
+open Bap.Std
+module Bap_value = Value
+open Bap_traces.Std
+open Monad.State
+open Bil.Result
+
+type event = Trace.event
+
+let mem_of_arch arch = 
+  let (module T : Target) = target_of_arch arch in
+  T.CPU.mem
+
+(** TODO: what is a behaviour of update_event method?  *)
+class context arch = object(self:'s)
+  inherit Bili.context
+  val mem  : var = mem_of_arch arch
+  val arch : arch = arch
+  method arch = arch
+  method mem = mem
+  method update_event: event -> 's = fun ev -> failwith "unimplemented"
+end
+
+let mem_operands mv = 
+  Size.of_int_opt (Bitvector.bitwidth (Move.data mv)) |> function
+  | None -> None
+  | Some sz ->
+    let addr = Bil.int (Move.cell mv) in
+    let data = Bil.int (Move.data mv) in
+    Some (addr, data, sz)
+
+class ['a] t = object(self)
```

Comment:
As I think it shouldn't. I don't think it is a good idea to have parameters in interpretator at all. 
It should just perform his work _under_ context, and a context in it's turn could have a paramters, instance varables and so on